pkgs/dartpad_ui/lib/main.dart

```diff
@@ -272,24 +273,27 @@ class _DartPadMainPageState extends State<DartPadMainPage>
     );
 
     appServices.populateVersions();
-    appServices
-        .performInitialLoad(
-            gistId: widget.gistId,
-            sampleId: widget.builtinSampleId,
-            flutterSampleId: widget.flutterSampleId,
-            channel: widget.initialChannel,
-            fallbackSnippet: Samples.defaultSnippet())
-        .then((value) {
-      // Start listening for inject code messages.
-      handleEmbedMessage(appServices, runOnInject: widget.runOnLoad);
-      if (widget.runOnLoad) {
-        appServices.performCompileAndRun();
-      }
-    });
+    _initAppServices();
```

Comment:
This function is only used once, so I would keep the code inlined here.

Reply:
Fair. I was editing that section anyway so I did that for readability -- it's flatter, the formatted is a little kinder, and it uses `async/await` instead of `.then` directly. 

I reverted that back to `.then()` though

Reply:
Why not make `initState` async?

Reply:
It's a Flutter thing: 

```console
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following assertion was thrown building Builder:
_DartPadMainPageState.initState() returned a Future.
State.initState() must be a void method without an `async` keyword.
Rather than awaiting on asynchronous work directly inside of initState, call a separate method to do
this work without awaiting it.
```

Reply:
I forgot about that rule, sorry for the churn. Let's keep it as is for now.